controllers/C_Document.class.php

```diff
@@ -158,6 +158,15 @@ function upload_action_process() {
         $sentUploadStatus = array();
         if (count($_FILES['file']['name']) > 0) {
             $upl_inc = 0;
+            
+            $targetFile = basename($_FILES['photo']['name']);
+            $uploadedFileType = pathinfo($targetFile,PATHINFO_EXTENSION);
+
+            // Allow certain file formats
+            if($uploadedFileType != "jpg" AND $uploadedFileType != "png" AND $uploadedFileType != "jpeg" AND $uploadedFileType != "gif" AND $uploadedFileType != "pdf" ){
+                die("Unsupported File Format, Please Try Again");
```

Comment:
This feature must allow a wide range of document types.  It is not about pictures.  It is for word processor files, pdf's, video formats, DICOM formats, html, and a million others.  The only thing we should be doing is disallowing certain executable extensions, and even those might have exceptions (unless compressed).